npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_error_handing.rb

```diff
@@ -0,0 +1,235 @@
+      "YN0060" => { message: "Incompatible peer dependency", error_class: Dependabot::DependencyFileNotResolvable },
+      "YN0061" => { message: "Deprecated package", error_class: Dependabot::DependabotError },
+      "YN0062" => { message: "Incompatible OS", error_class: Dependabot::DependabotError },
+      "YN0063" => { message: "Incompatible CPU", error_class: Dependabot::DependabotError },
+      "YN0068" => { message: "Unused package extension", error_class: Dependabot::DependabotError },
+      "YN0069" => { message: "Redundant package extension", error_class: Dependabot::DependabotError },
+      "YN0071" => { message: "NM can't install external soft link", error_class: Dependabot::DependabotError },
+      "YN0072" => { message: "NM preserve symlinks required", error_class: Dependabot::DependabotError },
+      "YN0074" => { message: "NM hardlinks mode downgraded", error_class: Dependabot::DependabotError },
+      "YN0075" => { message: "Prolog instantiation error", error_class: Dependabot::DependabotError },
+      "YN0076" => { message: "Incompatible architecture", error_class: Dependabot::DependencyFileNotResolvable },
+      "YN0077" => { message: "Ghost architecture", error_class: Dependabot::DependabotError },
+      "YN0078" => { message: "Resolution mismatch", error_class: Dependabot::DependencyFileNotResolvable },
+      "YN0080" => { message: "Network disabled", error_class: Dependabot::DependabotError },
+      "YN0081" => { message: "Network unsafe HTTP", error_class: Dependabot::DependabotError },
+      "YN0082" => { message: "Resolution failed", error_class: Dependabot::DependencyFileNotResolvable },
+      "YN0083" => { message: "Automerge git error", error_class: DependabotError },
+      "YN0085" => { message: "Updated resolution record", error_class: Dependabot::DependabotError },
+      "YN0086" => { message: "Explain peer dependencies CTA", error_class: Dependabot::DependabotError },
+      "YN0087" => { message: "Migration success", error_class: Dependabot::DependabotError },
+      "YN0088" => { message: "Version notice", error_class: Dependabot::DependabotError },
+      "YN0089" => { message: "Tips notice", error_class: Dependabot::DependabotError },
+      "YN0090" => { message: "Offline mode enabled", error_class: Dependabot::DependabotError }
+    }.freeze
+
+    NPM_REGISTERY = "registry.npmjs.org"
+
+    HTTP_CHECK_REGEX = %r{https?://}
+    TIMEOUT_FETCHING_PACKAGE_REGEX = %r{(?<url>.+)/(?<package>[^/]+): ETIMEDOUT}
+    UNREACHABLE_GIT_CHECK_REGEX = /ls-remote --tags --heads (?<url>.*)/
+
+    ONLY_PRIVATE_WORKSPACE_TEXT = "Workspaces can only be enabled in priva"
+    SUB_DEP_LOCAL_PATH_ERROR_TEXT = "refers to a non-existing file"
+    INVALID_PACKAGE = /Can't add "(?<package_req>.*)": invalid/
+
+    # Package not found error
+    MatchStruct = Struct.new(:text, :regex, :match_key, :split)
+
+    class HelperSubprocessFailed < Dependabot::SharedHelpers::HelperSubprocessFailed; end
```

Comment:
Created `HelperSubprocessFailed` under Dependabot::NpmAndYarn that will help to see all unknown erorrs under Dependabot::NpmAndYarn::HelperSubprocessFailed error without going into details. 

CC: @abdulapopoola , @jakecoffman , @jeffwidman